### PR TITLE
api: user: added email verification messages

### PR DESCRIPTION
**Pull request by gioelecerati**
**What does this pull request do? Explain your changes. (required)**

- Added text when signing in as unverified, asking to check for a verification email
- Added a text when signing up, informing that we sent a verification email
- Modified text for `email already registered` asking to sign in or check the verification email

**How did you test each of these updates (required)**

**Does this pull request close any open issues?**

**Screenshots (optional)**

**Checklist**

- [ ] I have read the **CONTRIBUTING** document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/api/src/controllers/user.ts [hunk 1]
```diff
@@ -409,6 +413,16 @@ app.post("/", validatePost("user"), async (req, res) => {
...
 
+  if (
+    req.config.requireEmailVerification &&
+    !user.emailValid &&
+    !user.admin &&
+    !req.config.isTest
+  ) {
+    res.status(403);
+    return res.json({ errors: ["we just sent you a verification email"] });
+  }
```
Comment (victorges) [inline packages/api/src/controllers/user.ts]: I don't think this check makes sense. Wasn't this user just created with `emailValid = validUser` var? Either verification will be disabled and this always passes, or verification is enabled and the registration request always fails. (this is the registration API right?)

File: packages/api/src/test-server.ts [hunk 2]
```diff
@@ -58,6 +58,7 @@ if (!params.insecureTestToken) {
 }
 params.listen = true;
 params.requireEmailVerification = true;
+params.isTest = true;
```
Comment (victorges) [inline packages/api/src/test-server.ts]: This is redundant, we should just use the `NODE_ENV` environment variable when we need to know if it is a test. e.g. https://github.com/livepeer/livepeer-com/blob/1fd399ff3387dd915a8b7a3ac36e123ce17af64b/packages/api/src/app-router.ts#L125-L126

**Review by victorges — APPROVED** [review]

File: packages/api/src/controllers/user.ts (hunk 1 above)
Comment (gioelecerati) [reply packages/api/src/controllers/user.ts]: yeah - it's failing a returning the text to be displayed under the form - similar to "email already registered" appearing just over the sign up button

I agree this should'nt be a 403 and a better way to display messages should exist

Comment (victorges) [reply packages/api/src/controllers/user.ts]: Definitely should not be a 403.

**Pull request merged**